consensus/tool/src/main.rs

```diff
@@ -0,0 +1,156 @@
+        })
+        .collect();
+
+    match config.tool_command {
+        ToolCommand::Status => {
+            for conn in &blockchain_conns {
+                let last_block_info = conn
+                    .get_last_block_info(&Empty::new())
+                    .expect("get last block info");
+
+                println!("{}", to_string_pretty(&json!({
+                    "index": last_block_info.index,
+                    "minimum_fees": Map::<String, Value>::from_iter(last_block_info.minimum_fees.into_iter().map(|(token_id, fee)| (token_id.to_string(), fee.into()))),
+                    "network_block_version": last_block_info.network_block_version,
+                })).expect("json error"))
+            }
+        }
+        ToolCommand::WaitForBlock { index, period } => loop {
+            let mut needs_retry = false;
+            for conn in &blockchain_conns {
+                match conn.get_last_block_info(&Empty::new()) {
+                    Ok(last_block_info) => {
+                        if last_block_info.index < index {
+                            needs_retry = true;
+                        }
+                    }
+                    Err(err) => {
+                        eprintln!("get_last_block_info(): {}", err);
+                        needs_retry = true;
+                    }
+                };
+            }
+            if needs_retry {
+                std::thread::sleep(Duration::from_secs(period));
+            } else {
+                break;
+            }
+        },
+        ToolCommand::WaitForQuiet {
+            period,
+            beyond_block,
+        } => {
+            let mut last_block_index = Option::<u64>::default();
+
+            let last_block_index = loop {
+                let mut was_updated = false;
+                for conn in &blockchain_conns {
+                    match conn.get_last_block_info(&Empty::new()) {
+                        Ok(last_block_info) => {
+                            if last_block_index != Some(last_block_info.index) {
+                                last_block_index = Some(last_block_info.index);
+                                was_updated = true;
+                            }
+                        }
+                        Err(err) => {
+                            eprintln!("get_last_block_info(): {}", err);
+                            was_updated = true;
+                        }
+                    };
+                }
+                if !was_updated {
+                    if let Some(last_block_index) = last_block_index.as_ref() {
+                        if beyond_block.is_none()
+                            || beyond_block.as_ref().unwrap() < last_block_index
+                        {
+                            break last_block_index;
+                        }
+                    }
+                }
+                std::thread::sleep(Duration::from_secs(period));
+            };
+            eprintln!(
```

Comment:
Nit: This message arguably belongs on stdout
```suggestion
            println!(
```